components/reference-implementation-openbanking-consumerdatastandards-au/src/main/java/org/wso2/openbanking/consumerdatastandards/au/extensions/handlers/CDSAuthorizationFlowHandler.java

```diff
@@ -0,0 +1,192 @@
+     * @param requestBody - the request body containing necessary parameters for consent screen population
+     * @return Response containing the consent and account data for the authorize screen
+     */
+    public static Response handlePopulateConsentAuthorizeScreen(PopulateConsentAuthorizeScreenRequestBody requestBody) {
+
+        try {
+            String requestId = requestBody.getRequestId();
+            PopulateConsentAuthorizeScreenData data = requestBody.getData();
+            Object requestParams = data.getRequestParameters();
+            String userId = requestBody.getData().getUserId();
+
+            //Convert request parameters to JSON object
+            JSONObject jsonRequestBody = CommonConsentExtensionUtils.convertObjectToJson(requestParams);
+
+            //Extract required data from request object
+            Map<String, Object> requiredData =
+                    ConsentAuthorizationUtil.extractRequiredDataFromRequestObject(jsonRequestBody);
+
+            //CDS consent retrieval step
+            SuccessResponsePopulateConsentAuthorizeScreenDataConsentData consentData =
+                    CDSDataRetrievalUtil.CdsConsentRetrieval(jsonRequestBody, requiredData);
+
+            //CDS account list retrieval step
+            SuccessResponsePopulateConsentAuthorizeScreenDataConsumerData consumerData =
+                    CDSDataRetrievalUtil.CdsConsumerDataRetrieval(jsonRequestBody, requiredData, userId);
+
+            //Set consent data to return to accelerator
+            SuccessResponsePopulateConsentAuthorizeScreenData screenData =
+                    new SuccessResponsePopulateConsentAuthorizeScreenData();
+            screenData.setConsentData(consentData);
+            screenData.setConsumerData(consumerData);
+
+            SuccessResponsePopulateConsentAuthorizeScreen response = new SuccessResponsePopulateConsentAuthorizeScreen();
+            response.setResponseId(requestId);
+            response.setStatus(SuccessResponsePopulateConsentAuthorizeScreen.StatusEnum.SUCCESS);
+            response.setData(screenData);
+
+            return Response.status(Response.Status.OK).entity(new JSONObject(response).toString()).build();
+
+        } catch (CDSConsentException ex) {
+            // Handle all CDS-related errors (including auto-converted exceptions)
+            log.error("CDS error during consent authorize screen population: " + ex.getMessage(), ex);
+
+            // Convert CDSConsentException to AuthorizationFailureException for consistent response format
+            AuthorizationFailureException authException = AuthorizationFailureException.createError(ex.getErrorEnum(),
+                ex.getErrorDetail(), CommonConstants.REJECTED_STATUS, requestBody.getRequestId());
+
+            return Response.status(Response.Status.OK).entity(authException.toFailedResponseJsonString()).build();
+
+        } catch (Exception e) {
```

Comment:
Can't we have a subset of exceptions we are expecting here?

Reply:
It is handled in fromThrowable method. Since there are 8 subsets I have moved it to a new method.